Automattic/wp-calypso
# Pricing page: make product card discount label smaller

### Changes proposed in this Pull Request

This PR makes the discount label in the Jetpack product card smaller, and moves it to the same line as the product name. The goal is to save space.

### Implementation notes

This PR does a couple of things:

- it moves the discount label out of the `DisplayPrice` component, to reflect the new structure
- it extracts the logic that computes the total discount price and percentage in the hook `useCouponDiscount`
- it adds tests for that hook

### Testing instructions

- Download the PR and run Calypso or cloud
- Visit the plans/pricing page
- Check that the discount labels are similar to the label in the capture below
- Check that the percentages are the same as in production for both the monthly and annual terms
- Check that the design of the card doesn't break with a long product name or label, as seen in the captures below
- Finally, check that the discount label is not showing for products you own or that are included in a plan you own

### Screenshots


<img width="372" alt="Screen Shot 2021-12-07 at 2 32 49 PM" src="https://user-images.githubusercontent.com/1620183/145094138-125f2560-3140-44bf-be90-a69d3db49559.png">
<img width="368" alt="Screen Shot 2021-12-07 at 2 34 11 PM" src="https://user-images.githubusercontent.com/1620183/145094292-0d802efa-8d19-4206-bf8e-9f14e63427c0.png">
<img width="368" alt="Screen Shot 2021-12-07 at 2 34 36 PM" src="https://user-images.githubusercontent.com/1620183/145094304-6887c8f7-369c-44ed-91c3-9104e54de32a.png">
